### PR TITLE
[CS-4259] [Tests] Add test coverage to Profile Purchase POC

### PR DESCRIPTION
**Pull request by oidoug**
### Description

Adds test coverage for hook `usePurchaseProfile`.

- [x] Completes #(CS-4259)

No visual updates.

### CODE REVIEW

File: cardstack/src/hooks/__tests__/usePurchaseProfile.test.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { act } from '@testing-library/react-native';
+import { useIAP, Product, Purchase } from 'react-native-iap';
+
+// import { CreateBusinessInfoDIDParams } from '@cardstack/types';
```
Comment (danibonilha) [inline cardstack/src/hooks/__tests__/usePurchaseProfile.test.ts]: letfover

File: cardstack/src/hooks/__tests__/usePurchaseProfile.test.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  accountAddress: '1234567890',
+};
+
+jest.mock('@rainbow-me/hooks', () => ({
+  useAccountProfile: jest.fn(),
+}));
+
+jest.mock('react-native-iap', () => ({
+  useIAP: jest.fn(),
+  // requestPurchase: mockedRequestPurchase,
```
Comment (danibonilha) [inline cardstack/src/hooks/__tests__/usePurchaseProfile.test.ts]: leftover

**Review by danibonilha — APPROVED** [review]
LGTM, just some leftovers, we could  however add a couple more of test cases, to check the `validateReceiptCreateProfile ` which I believe is pretty important, to check if after a valid purchase, the mutation is called or add a new task to handle it eventually

**Comment by oidoug** [comment]
@danibonilha , I'll try test the mutation before merging, I see we have something similar in `useRequestPrepaidCardScreen.test.ts`.

LMK if theres any other good example 👍

**Comment by oidoug** [comment]
@danibonilha , I've added a couple more tests to ensure `currentPurchase` triggers the verification, but didn't add tests to different mutation response statuses. I think that is valuable but still too soon to add, because the hook currently doesn't handle that either. I'll add more as we expand.

**Pull request merged**